cassiopeia/solver/MaxCutGreedySolver.py

```diff
@@ -0,0 +1,146 @@
+"""
+This file stores a subclass of GreedySolver, the MaxCutGreedySolver. The
+inference procedure here extends the "vanilla" Cassiopeia-Greedy, originally
+proposed in Jones et al, Genome Biology (2020). After each putative split of
+the samples generated by Cassioepia-Greedy, the hill-climbing prodecure from
+the MaxCutSolver is applied to the partition to optimize the it for the 
+max cut criterion on a connectivity graph built from the observed mutations 
+in the samples representing a supertree of phylogentic trees on each 
+individual character.
+"""
+import pandas as pd
+from typing import Callable, Dict, List, Optional, Tuple, Union
+
+from cassiopeia.solver import GreedySolver
+from cassiopeia.solver import graph_utilities
+from cassiopeia.solver.missing_data_methods import assign_missing_average
+
+
+class MaxCutGreedySolver(GreedySolver.GreedySolver):
+    """The MaxCutGreedySolver implements a top-down algorithm that recursively 
+    splits the sample set based on the presence/absence of the most frequent
+    mutation. Additionally, the hill-climbing procedure from the MaxCutSolver is 
+    used to further optimize each split for the max cut on the similarity graph 
+    on the samples. This effectively moves samples across the partition so that 
+    samples with similar mutations are grouped together and samples with 
+    different mutations are seperated. Multiple missing data imputation methods 
+    are included for handling the case when a sample has a missing value on the 
+    character being split, where presence or absence of the character is 
+    ambiguous. The user can also specify a missing data method.
+
+    Args:
+        character_matrix: A character matrix of observed character states for
+            all samples
+        missing_char: The character representing missing values
+        missing_data_classifier: Takes either a string specifying one of the
+            included missing data imputation methods, or a function
+            implementing the user-specified missing data method. The default is
+            the "average" method.
+        meta_data: Any meta data associated with the samples
+        priors: Prior probabilities of observing a transition from 0 to any
+            character state
+        fuzzy_solver: #TODO
```

Comment:
Don't leave a TODO in the arguments section of the code. The arguments should all do _something_. If they don't leave them out, and leave a TODO in the actual description of the class / function.